arangod/Graph/Providers/SingleServerProvider.h

```diff
@@ -65,9 +65,17 @@ struct SingleServerProvider {
    public:
     class Vertex {
      public:
-      explicit Vertex(VertexType v) : _vertex(v) {}
+      explicit Vertex(VertexType v) : _vertex(v) {
```

Comment:
The Vertex does not need the Depth.
Only the Step requires it.

Reply:
Moved to BaseStep and one level above ( -> Step)